### PR TITLE
(auto) formatting and cleanup fixes for test_volumes

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR fixes recent travis failures in travis job 5

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

run in lab env (similar to trillian)
```
Test Volume creation for all Disk Offerings (incl. custom) ... === TestName: test_01_create_volume | Status : SUCCESS ===
ok
Attach a created Volume to a Running VM ... === TestName: test_02_attach_volume | Status : SUCCESS ===
ok
Download a Volume attached to a VM ... === TestName: test_03_download_attached_volume | Status : SUCCESS ===
ok
Delete a Volume attached to a VM ... === TestName: test_04_delete_attached_volume | Status : SUCCESS ===
ok
Detach a Volume attached to a VM ... === TestName: test_05_detach_volume | Status : SUCCESS ===
ok
Download a Volume unattached to an VM ... === TestName: test_06_download_detached_volume | Status : SUCCESS ===
ok
Test resize (negative) non-existent volume ... === TestName: test_07_resize_fail | Status : SUCCESS ===
ok
Test resize a volume ... === TestName: test_08_resize_volume | Status : SUCCESS ===
ok
Delete a Volume unattached to an VM ... === TestName: test_09_delete_detached_volume | Status : SUCCESS ===
ok
test_10_list_volumes (test_volumes.TestVolumes) ... === TestName: test_10_list_volumes | Status : SUCCESS ===
ok
Attach a created Volume to a unstarted VM ... === TestName: test_11_attach_volume_with_unstarted_vm | Status : SUCCESS ===
ok
test_11_migrate_volume_and_change_offering (test_volumes.TestVolumes) ... === TestName: test_11_migrate_volume_and_change_offering | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 12 tests in 2755.733s

OK
```

### CODE REVIEW

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
🤞 Travis result

**Comment by DaanHoogland** [comment]
I ran a second test locally:
```
Test Volume creation for all Disk Offerings (incl. custom) ... === TestName: test_01_create_volume | Status : SUCCESS ===
ok
Attach a created Volume to a Running VM ... === TestName: test_02_attach_volume | Status : SUCCESS ===
ok
Download a Volume attached to a VM ... === TestName: test_03_download_attached_volume | Status : SUCCESS ===
ok
Delete a Volume attached to a VM ... === TestName: test_04_delete_attached_volume | Status : SUCCESS ===
ok
Detach a Volume attached to a VM ... === TestName: test_05_detach_volume | Status : SUCCESS ===
ok
Download a Volume unattached to an VM ... === TestName: test_06_download_detached_volume | Status : FAILED ===
FAIL
Test resize (negative) non-existent volume ... === TestName: test_07_resize_fail | Status : SUCCESS ===
ok
Test resize a volume ... === TestName: test_08_resize_volume | Status : SUCCESS ===
ok
Delete a Volume unattached to an VM ... === TestName: test_09_delete_detached_volume | Status : SUCCESS ===
ok
test_10_list_volumes (test_volumes.TestVolumes) ... === TestName: test_10_list_volumes | Status : SUCCESS ===
ok
Attach a created Volume to a unstarted VM ... === TestName: test_11_attach_volume_with_unstarted_vm | Status : SUCCESS ===
ok
# Validates the following ... === TestName: test_11_migrate_volume_and_change_offering | Status : EXCEPTION ===
ERROR
=== TestName: test_11_migrate_volume_and_change_offering | Status : EXCEPTION ===
ERROR

```
so looks like I only partly fixed it and am still dependent on circumstances :(

**Comment by DaanHoogland** [comment]
ah, just realised the failing tests above require hardware so won't run in travis.

**Comment by weizhouapache** [comment]
@DaanHoogland fyi, I run test_volumes.py twice without any issue. no code changes in main branch. I think travis test failures might not be related to  test_volumes.py .

**Comment by slavkap** [comment]
@DaanHoogland, again the volume isn't detached between tests `test_03_download_attached_volume` and `test_04_delete_attached_volume`. I guess the ssh connection timed out. That's why the tests have failed (my assumptions)

**Comment by DaanHoogland** [comment]
yes @weizhouapache I saw intermitted failures and successes as well as described above. Would be nice if we can make ci more relyable though :( .

File: test/integration/smoke/test_volumes.py [hunk 1]
```diff
@@ -383,29 +367,22 @@ def setUp(self):
...
-            self.virtual_machine.detach_volume(self.apiClient, self.volume)
-
         if self.virtual_machine.hypervisor == "KVM":
             self.virtual_machine.stop(self.apiClient)
+            if self.attached:
+                self.virtual_machine.detach_volume(self.apiClient, self.volume)
             self.virtual_machine.start(self.apiClient)
-            self.virtual_machine.get_ssh_client(reconnect = True)
-
+            self.virtual_machine.get_ssh_client(reconnect=True)
```
Comment (slavkap) [inline test/integration/smoke/test_volumes.py:375]:
```suggestion
            try:
                self.virtual_machine.get_ssh_client(reconnect=True)
            except Exception as err:
                self.fail("SSH failed for Virtual machine: %s due to %s" %
                      self.virtual_machine.ipaddress, err)
```
To catch it if it fails here

**Comment by DaanHoogland** [comment]
@slavkap @rhtyd in a real environment it runs fine. I think this job 5 failure is either simulator or travis specific.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by slavkap** [comment]
@DaanHoogland now job 5 has passed [https://travis-ci.com/github/apache/cloudstack/builds/230665717](https://travis-ci.com/github/apache/cloudstack/builds/230665717)
I'll run the tests later today on my dev. env.

**Comment by rhtyd** [comment]
Travis passing, smoketests LGTM - failures don't seem related to this PR.

**Pull request merged**